subcommands/keys/tuf_updates_delete_offline_key.go

```diff
@@ -0,0 +1,120 @@
+  fioctl keys tuf updates delete-offline-key \
+    --txid=abc --role=targets
+	 --key-id=15bbb6e79c9ac73b2db7df73c96f3a4937a25d948c048ba0208e49e426e5888a`,
+		Run: doTufUpdatesDeleteOfflineKey,
+	}
+	del.Flags().StringP("role", "r", "", "TUF role name, supported: Root, Targets.")
+	_ = del.MarkFlagRequired("role")
+	del.Flags().StringP("txid", "x", "", "TUF root updates transaction ID.")
+	del.Flags().StringP("keys", "k", "", "Path to <tuf-root-keys.tgz> used to sign TUF root.")
+	_ = del.MarkFlagFilename("keys")
+	del.Flags().StringP("key-id", "i", "", "A key ID to delete, as specified in your TUF root.")
+	del.MarkFlagsMutuallyExclusive("keys", "key-id")
+	tufUpdatesCmd.AddCommand(del)
+}
+
+func doTufUpdatesDeleteOfflineKey(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	txid, _ := cmd.Flags().GetString("txid")
+	keysFile, _ := cmd.Flags().GetString("keys")
+	keyId, _ := cmd.Flags().GetString("key-id")
+	roleName, _ := cmd.Flags().GetString("role")
+	roleName = ParseTufRoleNameOffline(roleName)
+
+	var (
+		creds OfflineCreds
+		err   error
+	)
+
+	if keysFile != "" {
+		creds, err = GetOfflineCreds(keysFile)
+		subcommands.DieNotNil(err)
+	} else if keyId == "" {
+		subcommands.DieNotNil(errors.New(
+			"Either --keys or --key-id option is required to delete the offline TUF key.",
+		))
+	}
+
+	updates, err := api.TufRootUpdatesGet(factory)
+	subcommands.DieNotNil(err)
+	_, newCiRoot := checkTufRootUpdatesStatus(updates, true)
+
+	var (
+		roleToUpdate *tuf.RootRole
+		validKeyIds  []string
+	)
+
+	switch roleName {
+	case tufRoleNameRoot:
+		roleToUpdate = newCiRoot.Signed.Roles["root"]
+		validKeyIds = roleToUpdate.KeyIDs
+	case tufRoleNameTargets:
+		roleToUpdate = newCiRoot.Signed.Roles["targets"]
+		onlineTargetsId := updates.Updated.OnlineKeys["targets"]
+		if onlineTargetsId == "" {
+			subcommands.DieNotNil(errors.New("Unable to find online target key for factory"))
+		}
+		if keyId != "" && keyId == onlineTargetsId {
+			subcommands.DieNotNil(fmt.Errorf(
+				"It is not allowed to delete an online TUF targets key: %s", onlineTargetsId,
+			))
+		}
+		validKeyIds = subcommands.SliceRemove(roleToUpdate.KeyIDs, onlineTargetsId)
+	default:
+		panic(fmt.Errorf("Unexpected role name: %s", roleName))
+	}
+
+	if keyId == "" {
+		oldKey, err := FindOneTufSigner(newCiRoot, creds, validKeyIds)
+		subcommands.DieNotNil(err)
+		keyId = oldKey.Id
+	} else if !slices.Contains(validKeyIds, keyId) {
+		subcommands.DieNotNil(fmt.Errorf(
+			"Key ID %s not found in TUF %s role keys: %v", keyId, roleName, validKeyIds,
+		))
+	}
+	roleToUpdate.KeyIDs = subcommands.SliceRemove(roleToUpdate.KeyIDs, keyId)
+
+	fmt.Println("= Delete keyid:", keyId)
+	newCiRoot.Signatures = make([]tuf.Signature, 0)
+	removeUnusedTufKeys(newCiRoot)
+	newProdRoot := genProdTufRoot(newCiRoot)
+
+	fmt.Println("= Uploading new TUF root")
+	subcommands.DieNotNil(api.TufRootUpdatesPut(factory, txid, newCiRoot, newProdRoot, nil))
+}
```

Comment:
So, we don't delete the key file, just remove the key from root meta?

Reply:
ok.
How a user can list IDs of keys located in a single key file? For example, if a user decides to delete one of the old keys, or actually add it again to root meta :). I think we forbid it for purpose since it sounds too hackish :).

Reply:
> How a user can list IDs of keys located in a single key file?

I do `vim keys.tar.gz` for that... only because I'm too lazy to remember a proper `tar -tf keys.tar.gz` (or what) command.
For offline keys, a key ID is a part of a file name.

Reply:
>or offline keys, a key ID is a part of a file name.

I meant the case when a single key file has many keys, one active (present in root) and a few older/rotated.

If we delete a key with ID=`id1` from a key file `id1.tgz` and then add a new key with ID=`id2` then the fioctl command will create a new key file `id2.tgz` or rename `id1.tgz` to `id2.tgz`?

Reply:
Sorry, you misread my comment because I gave not enought details.

A typical `keys.tgz` file has an inner folder structure, where the leaf nodes have names like `tuf-root-<id1>.pub, tuf-root-<id1>.sec, tuf-root-<id2>.pub, tuf-root-<id2>.sec, ...`. A simple way to get a list of key IDs is to list file names inside a tarball.

Reply:
ok. This is a file name inside the TAR, not the *.tgz file.